test/functional/test_nic_stat_error_inject.py

```diff
@@ -0,0 +1,209 @@
+        "execute": "qmp_capabilities"
+    }
+
+    s.send(json.dumps(payload_enable_qmp))
+    s.recv()
+    # wait until system is ready for ssh.
+    ssh = sshclient.SSH(host="127.0.0.1", username="root", password="root", port=2222)
+    ssh.wait_for_host_up()
+
+
+def stop_node():
+    global s
+    global conf
+    s.close()
+    node = model.CNode(conf)
+    node.init()
+    node.stop()
+    node.terminate_workspace()
+    conf = {}
+
+
+def get_interface_name():
+    global inface
+    global ssh
+    with open("ifg_info.txt", "w+") as ff:
+        status, output = ssh.exec_command("ifconfig")
+        ff.write(output)
+        ff.seek(0, 0)
+        for str0 in ff.readlines():
+            if "UP,BROADCAST,RUNNING,MULTICAST" in str0:
+                str1 = str0.split()
+                # in order to delete ":"
+                inface = str1[0][0:-1]
+                break
+        ff.close
+
+
+class nicdata_check:
+    Grxstat = 0
+    Grxerr = 0
+    Gtxstat = 0
+    Gtxerr = 0
+
+    def getnicdata(self):
+        status, output = ssh.exec_command('ifconfig -s {}'.format(inface))
+        with open("ifg_s.txt", "w+") as fd:
+            fd.write(output)
+            fd.seek(0, 0)
+            for line in fd.readlines():
+                print(line)
+                line1 = line.split()
+                if line1[0] == inface:
+                    self.Grxstat = long(line1[2])
+                    self.Grxerr = long(line1[3])
+                    self.Gtxstat = long(line1[6])
+                    self.Gtxerr = long(line1[7])
+            fd.close()
+
+
+class test_nic_error_inject(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        start_node()
+
+    @classmethod
+    def tearDownClass(cls):
+        stop_node()
+
+    def test_status_code_error_inject(self):
+        status_code_error = {
+            "execute": "vmxnet3-inject-stats-error",
+            "arguments": {
+                "id": "mgmt",
+                "dir": None,
+                "ratio": None
+            }
+        }
+        status_code_error_full = {
+            "execute": "vmxnet3-inject-stats-error",
+            "arguments": {
+                "id": "mgmt",
+                "dir": "full",
+                "ratio": 1
+            }
+        }
+        status_code_meter = [
+            ["tx", 0.2],
+            ["rx", 0.1],
+        ]
+
+        get_interface_name()
+        # first get raw data
+        fdata = nicdata_check()
+        fdata.getnicdata()
+        self.assertEqual(0, fdata.Gtxerr, "first txerr not 0")
+        self.assertEqual(0, fdata.Grxerr, "first rxerr not 0")
+
+        # second change ratio
+        for status_code_meters in status_code_meter:
+            status_code_error["arguments"]["dir"] = status_code_meters[0]
+            status_code_error["arguments"]["ratio"] = status_code_meters[1]
+            s.send(json.dumps(status_code_error))
+            count = 1
+            while count < 2:
```

Comment:
why try 2 counts here?

Reply:
avoid first fail

Reply:
the initial count value is 1, in your conditional statement, if the first "try" succeeds, this will be end in an infinite loop, otherwise the count will be increased to 2, and it will not meet the condition "count < 2" and break out of loop. In this case I think the "assert" check takes no effect.

Reply:
If the recv packets count is uncertain, i.e., it could be 1 or 2, you can add a "break" in try block. If it is surely 2, just receive twice and check the outputs.

Reply:
get it ,thanks